delete_expired_objects/delete_expired_objects_job.py

```diff
@@ -1,32 +1,49 @@
 from time import time
 from gen3 import object, metadata, auth
 
+"""
+Note: This job requires Metadata service to be above version 1.6.4
+"""
 auth = auth.Gen3Auth(refresh_file="/mnt/api-profile-credentials.json")
 mds_handle = metadata.Gen3Metadata(auth)
 
-print("Initializing delete_temp_objects_job...")
-LIMIT_SIZE = 10
+print("Initializing delete_expired_objects_job...")
+
+LIMIT_SIZE = 2000
 offset_position = 0
 response_dict = mds_handle.query(
     query="date_to_delete=*", return_full_metadata=True, limit=LIMIT_SIZE
 )
-if type(response_dict) is dict:
-    guid_list = list(response_dict.values())
-    while len(response_dict) == LIMIT_SIZE:
-        offset_position += LIMIT_SIZE
-        response_dict = mds_handle.query(
-            query="date_to_delete=*",
-            return_full_metadata=True,
-            limit=LIMIT_SIZE,
-            offset=offset_position,
-        )
-        guid_list += list(response_dict.values())
-    delete_list = [obj["guid"] for obj in guid_list if obj["date_to_delete"] < time()]
-    object_api = object.Gen3Object(auth)
-    for guid in delete_list:
+
+if type(response_dict) is not dict:
+    raise Exception(f"Job Failed. Couldn't fetch records. -- {response_dict}")
+
+
+guid_list = list(response_dict.values())
+while len(response_dict) == LIMIT_SIZE:
+    offset_position += LIMIT_SIZE
+    response_dict = mds_handle.query(
+        query="date_to_delete=*",
+        return_full_metadata=True,
+        limit=LIMIT_SIZE,
+        offset=offset_position,
+    )
+    guid_list += list(response_dict.values())
+exception_counter, delete_counter = 0, 0
+object_api = object.Gen3Object(auth)
+
+for obj in list(response_dict.values()):
+    if obj["date_to_delete"] < time():
         try:
-            print(f"Deleting object with guid -- {guid}")
-            object_api.delete_object(guid=guid)
+            print(f"Deleting object with guid -- {obj['guid']}")
+            object_api.delete_object(guid=obj["guid"], delete_file_locations=True)
+            delete_counter += 1
         except Exception as ex:
-            print(f"Couldn't delete object with guid -- {guid}\n{ex}")
-print(f"Job completed. Deleted {len(delete_list)} objects...")
+            exception_counter += 1
+            print(f"Couldn't delete object with guid -- {obj['guid']}\n{ex}")
```

Comment:
instead of `print(ex)` we could use `traceback.print_exc()` to log the traceback for easier debugging